Build failure: discount on aarch64-darwin

### Steps To Reproduce

Steps to reproduce the behavior:
1. `nix run nixpkgs#discount` on aarch64-darwin

### Build log

```
error: builder for '/nix/store/25mf9akcx9hsi8wf3gnh0nxqcibyn102-discount-2.2.7b.drv' failed with exit code 2;
       last 10 log lines:
       >                                             ^
       > 1 error generated.
       > generate.c:629:1: warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
       > p_or_nothing(p)
       > ^
       > clang -Wno-implicit-int -I. -g -fPIC -c -o css.o css.c
       > make: *** [Makefile:127: main.o] Error 1
       > make: *** Waiting for unfinished jobs....
       > 8 warnings generated.
       > 1 warning generated.
       For full logs, run 'nix log /nix/store/25mf9akcx9hsi8wf3gnh0nxqcibyn102-discount-2.2.7b.drv'.
```

### Additional context

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.2.0, macOS 14.2.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/nix/store/dllvbhgw5f584ficlqvqkpb0mjnf4v0s-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/text/discount/default.nix
+++ b/pkgs/tools/text/discount/default.nix
@@ -1,14 +1,14 @@
 { lib, stdenv, fetchFromGitHub }:
 
 stdenv.mkDerivation rec {
-  version = "2.2.7b";
+  version = "3.0.0d";
   pname = "discount";
 
   src = fetchFromGitHub {
     owner = "Orc";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-S6OVKYulhvEPRqNXBsvZ7m2W4cbdnrpZKPAo3SfD+9s=";
+    sha256 = "sha256-fFSlW9qnH3NL9civ793LrScOJSuRe9i377BgpNzOXa0=";
   };
 
   patches = [ ./fix-configure-path.patch ];

